### ISSUE TITLE
'Related Products' not working

### ISSUE THREAD

**Issue by janetgatsby** [patpluspun/fmp_au#45]
Hi Janet

 

I have noticed that ‘Related Products’ is not working.

 

When I’m adding a title to the website and I add a book to the ‘Related Products’ section, it is not showing on the live website.

 

Please see example attached and below, The Rig appears in the ‘Related Products’ section in the Content Management System, but is not appearing on the live website (see link).

 

The only way I could get it to appear on the live site is to add it to the ‘We Also Recommend’ section. It should only be in ‘Related Products’.  

http://www.fivemile.com.au/catalog/new-releases/crystal-force

 

Could you please add this to the list of issues to be fixed?

 

Many thanks
![related products](https://cloud.githubusercontent.com/assets/7650441/6998666/30447226-dbb8-11e4-9481-09dd1845faa9.jpg)

**Comment by janetgatsby** [comment]
@patpluspun - this issue needs attention, it is lonely.

**Comment by janetgatsby** [comment]
@patpluspun this issue is still lonely and will soon be posting a personal ad to get the attention it so craves.

**Comment by patpluspun** [comment]
I'm on it now.  I will update when I have a fix.

**Comment by patpluspun** [comment]
This was never actually implemented, I think the inline description on the field indicated that.  I went ahead and added a new block to output related products, it is available here:

https://github.com/patpluspun/fmp_au/pull/51

It needs a touch of theming, I'll rope in @zombiesundays to get that done, shouldn't take very long at all.

1 hour.

**Comment by janetgatsby** [comment]
@patpluspun this one doesn't seem to be working on prod. I can see that on http://www.fivemile.com.au/catalog/new-releases/crystal-force "Crystal Force" the related product is set to "The Rig". However on the website under Related Product it is displaying "Crystal Force" aka it's related to itself!

**Comment by patpluspun** [comment]
I'll pull this down to dev and look at it.  I'll make a hotfix tonight to fix it.

**Comment by janetgatsby** [comment]
Did you find out anything?

**Comment by patpluspun** [comment]
No, the only thing I can see is that the view that is powering it has changed, likely due to updating the views module.  I really, really hate views, but it would be a complete chore to remove it at this point...

I'm working on a workaround for now.

**Comment by patpluspun** [comment]
This is utterly baffling, all the rest of the views work just fine, but something about the Related Products field is making it not work.  I can try deleting it and re-adding it, but any content that's been stored there will be lost.  I will duplicate the Recommended field, since that one is working as expected.

**Comment by patpluspun** [comment]
Ok @janetgatsby I have worked around it, and the output doesn't use views at all anymore (and now I'm ready to start stripping views out completely).  I have updated test.fivemile.com.au with the hotfix branch, but it is in serious need of theming.  We're gonna need @zombiesundays to come in and fix her up before I can deploy this.  I have left my dev server on the same branch though, so once he's available I can get him in and he can do his magic and we can get this out the door.  You can test for functionality while we wait on him though.

**Comment by janetgatsby** [comment]
Hey Patrick, @zombiesundays told me he is a bit busy these days and suggested I call Simon Figeuroa to help out. Do you know him? 

I'll contact him this weekend and get him set up, could you get him access to Github plus dev and test? simon.figeuroa88@gmail.com

**Comment by patpluspun** [comment]
This is ready for test: http://test.fivemile.com.au/catalog/new-releases/test-ebook-kobo-link
I noticed the "Other books in this series" area is broken too, I will address that in another ticket though as it will need to be converted to the new way of handling these queries for ease of theming.

2 hours.

**Comment by janetgatsby** [comment]
@patpluspun tested and approved for Related Products functionality.
